src/views/directMessages/components/messageThreadListItem.js

```diff
@@ -1,19 +1,65 @@
 import React, { Component } from 'react';
+import compose from 'recompose/compose';
+import { connect } from 'react-redux';
 // $FlowFixMe
+import { addToastWithTimeout } from 'src/actions/toasts';
 import Link from 'src/components/link';
 import { timeDifference } from '../../../helpers/utils';
 import { renderAvatars } from './avatars';
+import archiveDirectMessageThreadMutation from 'shared/graphql/mutations/directMessageThread/archiveDirectMessageThread';
+import unarchiveDirectMessageThreadMutation from 'shared/graphql/mutations/directMessageThread/unarchiveDirectMessageThread';
 import {
+  ArchiveUnarchiveCTA,
   Wrapper,
   Row,
   Meta,
   MessageGroupTextContainer,
   MessageGroupByline,
   Usernames,
   Timestamp,
+  GearButton,
 } from './style';
 
 class ListCardItemDirectMessageThread extends Component {
+  handleGearClick = e => {
+    e.stopPropagation(); // We need this since the whole wrapper is clickable
+    e.preventDefault();
+  };
+
+  handleArchiveDMThread = e => {
+    e.stopPropagation();
+    e.preventDefault();
+    const { thread: { id: threadId }, dispatch } = this.props;
+
+    this.props
+      .archiveDirectMessageThread(threadId)
+      .then(({ data }) => {
```

Comment:
@mxstbr I should update the `UserDMThreads` after this operation, but the _refresh_ is not happening. My guess is that I'm calling this mutation in `src/views/directMessages/components/messageThreadListItem.js` instead of `src/views/directMessages/containers/index.js`.
Is that correct?